spec.md

```diff
@@ -1354,6 +1370,14 @@ message ControllerServiceCapability {
       // with the snapshot_id as the filter to query whether the
       // uploading process is complete or not.
       LIST_SNAPSHOTS = 6;
+      // Plugins supporting volume cloning at the storage level MAY
+      // report this capability. The source volume must be managed by 
+      // the same plugin.  Not all volume sources and parameters
+      // combinations may work. Plugins MUST NOT implement this feature
+      // by attaching 2 volumes and copying the content, but COs may
+      // implement it that way for plugins that don't support this
+      // feature.
```

Comment:
We should just drop:

```
      // Plugins MUST NOT implement this feature
      // by attaching 2 volumes and copying the content, but COs may
      // implement it that way for plugins that don't support this
      // feature.
```

Seems like an implementation detail. Good plugins won't do it, we don't need to spell it out.

Reply:
I'll remove it, but after the "efficient cloning" discussion on this review, we decided in last weeks meeting (if I understood it right) to spell it out in the comments to ensure that plugins and COs were clear on this point, and users could expect a consistent behavior across plugins: cloning is a fast operation.

Reply:
> I'll remove it, but after the "efficient cloning" discussion on this PR, we decided in last weeks meeting (if I understood it right) to spell it out in the comments to ensure that plugins and COs were clear on this point, and users could expect a consistent behavior across plugins: cloning is a fast operation.

We did talk about some comments but I think having it in the form of "MUST NOT" is too strong here, and it is an implementation detail so if someone wants to do it then hey, that's up to them.

Reply:
> We did talk about some comments but I think having it in the form of "MUST NOT" is too strong here, and it is an implementation detail so if someone wants to do it then hey, that's up to them.

That is the situation I wanted to avoid, unmet expectations of clone on some plugins/backends.  If a backend doesn't support efficient cloning and they report they do, then they are not following the specs, and users can call them out on it.

But maybe it's not such a big deal to people as I'm making it to be.

Reply:
Meh... I don't know.  Anyway, LGTM; I'm still not a contributor so you just get my moral support  in the form of a +1